vaadin-crud-flow-parent/vaadin-crud-testbench/src/main/java/com/vaadin/flow/component/crud/testbench/CrudElement.java

```diff
@@ -81,7 +81,7 @@ public void openRowForEditing(int row) {
         if (isEditOnClick()) {
             this.getGrid().getCell(row, 0).click();
         } else {
-            this.$("vaadin-crud-edit").all().get(row).click();
+            this.$("vaadin-crud-edit").all().get(row + 1).click();
```

Comment:
This is flaky, as it relies on the order in which the "vaadin-crud-edit" elements appear in the light DOM of grid and it is in no way deterministic! Even the previous solution had that flaw.

Reply:
I wonder if there would be a cleaner solution to first accessing the correct row by the given index and then the edit element on that row.

What really makes me wonder the most is that why the "sizer" row has the edit element in the first place. All the cells in the sizer should be empty. Verified this in the crud [demo](https://cdn.vaadin.com/vaadin-crud/1.3.0/demo/#crud-basic-demos).

Reply:
Checking the [Java demos](https://vaadin.com/components/vaadin-crud/java-examples), there might be some difference how it works with Flow.

The `GridElement#getRow` method also relies on the DOM to get the `<tr>` element: 
```js
var rowsInDom = grid.$.items.children;
var rowInDom = Array.from(rowsInDom).filter(function(row) { 
  return !row.hidden && row.index == rowIndex;
})[0];
```

Reply:
That doesn't rely on the DOM index, but it actually checks the `row.index` property, which is the correct approach

Reply:
> Checking the Java demos, there might be some difference how it works with Flow

Seems so. The sizer cells should never have any content.

Reply:
Ah, okay. My bad.

Couldn't find a nice way to go from the row, since from `GridElement#getRow` we are on the Grid's shadow DOM and need to go back to the light DOM to find the `vaadin-crud-edit` element:
```js
// $0 is the <tr> element
Array.from($0.cells)
  .map(cell => cell.firstElementChild.assignedNodes()[0].children[0])
  .filter(el => el && el.localName == 'vaadin-crud-edit')[0]
```

Reply:
Have tried to go with cleaner solution to get `vaadin-crud-edit`. It has became a bit longer, but now it's based on `row.index`.

As for the sizer cells content, I guess we can create another issue 🤔

Reply:
> As for the sizer cells content, I guess we can create another issue

Sounds good. It is a separate bug.